code/scripts/templates/monitrc

```diff
@@ -0,0 +1,321 @@
+## see https://mmonit.com/). By default Monit registers credentials with
+## M/Monit so M/Monit can smoothly communicate back to Monit and you don't
+## have to register Monit credentials manually in M/Monit. It is possible to
+## disable credential registration using the commented out option below.
+## Though, if safety is a concern we recommend instead using https when
+## communicating with M/Monit and send credentials encrypted. The password
+## should be URL encoded if it contains URL-significant characters like
+## \":\", \"?\", \"@\".
+#
+#  set mmonit http://monit:monit@104.46.37.155:2810/collector
+#     # and register without credentials     # Don't register credentials
+#
+#
+## Monit by default uses the following format for alerts if the the mail-format
+## statement is missing::
+## --8<--
+## set mail-format {
+##   from:    Monit <monit@$HOST>
+##   subject: monit alert --  $EVENT $SERVICE
+##   message: $EVENT Service $SERVICE
+##                 Date:        $DATE
+##                 Action:      $ACTION
+##                 Host:        $HOST
+##                 Description: $DESCRIPTION
+##
+##            Your faithful employee,
+##            Monit
+## }
+## --8<--
+   set mail-format {
+       from: torqmonit@gmail.com 
+    subject: [\$SERVICE] monit alert -- \$EVENT at \$DATE
+    message: Monit Report: 
+             ACTION:          \$ACTION 
+            SERVICE:          \$SERVICE 
+               DATE:          \$DATE
+               HOST:          \$HOST 
+        DESCRIPTION:          \$DESCRIPTION
+             
+       Powered by Monit
+  
+       This message has been generated automatically! 
+   }
+##
+## You can override this message format or parts of it, such as subject
+## or sender using the MAIL-FORMAT statement. Macros such as $DATE, etc.
+## are expanded at runtime. For example, to override the sender, use:
+#
+# set mail-format { from: monit@foo.bar }
+#
+#
+## You can set alert recipients whom will receive alerts if/when a
+## service defined in this file has errors. Alerts may be restricted on
+## events by using a filter as in the second example below.
+#
+# set alert sysadm@foo.bar                       # receive all alerts
+#
+## Do not alert when Monit starts, stops or performs a user initiated action.
+## This filter is recommended to avoid getting alerts for trivial cases.
+#
+# set alert your-name@your.domain not on { instance, action }
+#
+#
+## Monit has an embedded HTTP interface which can be used to view status of
+## services monitored and manage services from a web interface. The HTTP
+## interface is also required if you want to issue Monit commands from the
+## command line, such as 'monit status' or 'monit restart service' The reason
+## for this is that the Monit client uses the HTTP interface to send these
+## commands to a running Monit daemon. See the Monit Wiki if you want to
+## enable SSL for the HTTP interface.
+#
+ set httpd port 2812 and
```

Comment:
anyway to have this port based on TorQ baseport variable? what happens if a user already has something running on port 2812?

Reply:
this is the default port for monit. It wasn't my focus to change it, but it should be really easy to do so. monitrc is created using the environment variables which are set when `setenv.sh` is executed. Should I look at a way to set that port taking in consideration the `baseport` and the number of processes or should I maybe define a an env variable and let the user set the port (just like we do now with the `baseport`)

Reply:
personally I think the less a user has to think about configuring the better, so tying it to the baseport seems most logical to me